File: SEM5/MobileApp/lab5_server/expense_tracker/expense_tracker_last/expense_tracker_server/expense.py
class Expense:
    def __init__(self, id, amount: float, type: str, date: str, note: str):
        self.id = id
        self.amount = amount
        self.type = type
        self.date = date
        self.note = note


    def as_dict(self):
        return {
            "id": self.id,
            "amount": self.amount,
            "type": self.type,
            "date": self.date,
            "note": self.note
        }

File: SEM5/MobileApp/lab5_server/expense_tracker/expense_tracker_last/expense_tracker_client/lib/models/expense.dart
class Expense{
  late int id;
  double amount;
  String type;
  String date;
  String note;

  Expense(this.amount, this.type, this.date, this.note);

  Map<String, dynamic> toMapWithId(){
    return {
      'id': id,
      'amount': amount,
      'type': type,
      'date': date,
      'note': note
    };
  }

  Map<String, dynamic> toMap(){
    return {
      'amount':amount,
      'type': type,
      'date': date,
      'note': note
    };
  }

  factory Expense.fromMap(Map<String, dynamic> map){
    var expense = Expense(map['amount'], map['type'], map['date'], map['note']);
    expense.id = map['id'];
    return expense;
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'amount': amount,
      'type': type,
      'date': date,
      'note': note
    };
  }

  factory Expense.fromJson(Map<String, dynamic> map) {
    var expense = Expense(map['amount'], map['type'], map['date'], map['note']);
    expense.id = map['id'];
    return expense;
  }
}